Sentinel/Classes/Core/PreferencesInfo/PreferenceItem.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  OptionGenericItem.swift
+//  Sentinel
+//
+//  Created by Infinum on 04.09.2023..
+//
+
+import UIKit
+
+@objcMembers
+public class PreferenceItem: NSObject {
+
+    enum InputError: LocalizedError {
+        case formatError
+        case outOfRange
+        case enumCaseNotFound
+    }
+
+    /// Input type of preference. Defines what UI control to display.
+    enum InputType {
+        /// String input
+        case text
+        /// Boolean input: "true" or "false"
+        case bool
+        /// Integer input with minimum and maximum values
+        case integer(min: Int?, max: Int?)
+        /// Double input with minimum and maximum values
+        case double(min: Double?, max: Double?)
+        /// Enum input, as a RawRepresentable with either String or Int as type.
+        case enumeration(allCases: [any RawRepresentable])
```

Comment:
I thought about having a simple array of strings here as input type. But enums in swift carry more information - both their label and their raw value, e.g if for `case north = "N"` I can display `"north (N)"`.

Reply:
For now only single selection is possible. Here you are supposed to provide all the selectable cases of an enum.